scripts/build/builders/realtek.py

```diff
@@ -0,0 +1,171 @@
+        elif self == RealtekApp.WINDOW:
+            return 'window-app'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+    @property
+    def TargetName(self):
+        if self == RealtekApp.LIGHT:
+            return 'matter-cli-ftd'
+        elif self == RealtekApp.LIGHT_SWITCH:
+            return 'matter-cli-mtd'
+        elif self == RealtekApp.LOCK:
+            return 'matter-cli-mtd'
+        elif self == RealtekApp.WINDOW:
+            return 'matter-cli-mtd'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+    @property
+    def AppNamePrefix(self):
+        if self == RealtekApp.LIGHT:
+            return 'chip-rtl8777g-lighting-app'
+        elif self == RealtekApp.LIGHT_SWITCH:
+            return 'chip-rtl8777g-light-switch-app'
+        elif self == RealtekApp.LOCK:
+            return 'chip-rtl8777g-lock-app'
+        elif self == RealtekApp.WINDOW:
+            return 'chip-rtl8777g-window-app'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+
+class RealtekBuilder(Builder):
+
+    def __init__(self,
+                 root,
+                 runner,
+                 board: RealtekBoard = RealtekBoard.RTL8777G,
+                 app: RealtekApp = RealtekApp.LIGHT,
+                 enable_cli: bool = False,
+                 enable_rpc: bool = False,
+                 enable_shell: bool = False):
+        super(RealtekBuilder, self).__init__(root, runner)
+        self.board = board
+        self.app = app
+        self.enable_cli = enable_cli
+        self.enable_rpc = enable_rpc
+        self.enable_shell = enable_shell
+
+        os.environ['OT_SRCDIR'] = os.path.join(os.getcwd(), 'third_party/openthread/ot-realtek')
+        os.environ['REALTEK_SDK_PATH'] = os.path.join(os.environ['OT_SRCDIR'], 'third_party/Realtek/rtl87x2g_sdk')
+        os.environ['BUILD_BANK'] = 'bank0'
+        os.environ['OT_CMAKE_NINJA_TARGET'] = self.app.TargetName
+
+    def CmakeBuildFlags(self):
+        flags = []
+
+        flags.append("-DCMAKE_BUILD_TYPE=Release")
+        flags.append("-DCMAKE_TOOLCHAIN_FILE=src/bee4/arm-none-eabi.cmake")
+        flags.append("-DBUILD_TYPE=sdk")
+        flags.append(f"-DBUILD_TARGET={self.board.BoardName}")
+        flags.append(f"-DBUILD_BOARD_TARGET={self.board.BoardName}")
+        flags.append(f"-DOT_CMAKE_NINJA_TARGET={self.app.TargetName}")
+
+        if self.enable_cli:
+            flags.append("-DENABLE_CLI=ON")
+        else:
+            flags.append("-DENABLE_CLI=OFF")
+
+        if self.enable_rpc:
+            flags.append("-DENABLE_PW_RPC=ON")
+        else:
+            flags.append("-DENABLE_PW_RPC=OFF")
+
+        if self.enable_shell:
+            flags.append("-DENABLE_SHELL=ON")
+        else:
+            flags.append("-DENABLE_SHELL=OFF")
+
+        build_flags = " ".join(flags)
+
+        return build_flags
+
+    def generate(self):
+        os.environ['OUT_FOLDER'] = self.output_dir
```

Comment:
why is this setting environment rather than passing it to cmd?  MATTER_EXAMPLE_PATH seems to be exported.

Do we have all these environment variables documented somewhere?